packages/internal/src/diagnostics/index.ts

```diff
@@ -0,0 +1,137 @@
+import { Project, ts } from 'ts-morph';
+import { init, getGraphQLDiagnostics } from '@0no-co/graphqlsp/api';
+import path from 'path';
+import fs from 'fs';
+import { resolveTypeScriptRootDir } from '../resolve';
+import type { SchemaOrigin } from '../loaders';
+import { load } from '../loaders';
+
+type Severity = 'error' | 'warn' | 'info';
+const severities: Severity[] = ['error', 'warn', 'info'];
+export interface FormattedDisplayableDiagnostic {
+  severity: Severity;
+  message: string;
+  line: number;
+  col: number;
+  file: string | undefined;
+}
+
+type GraphQLSPConfig = {
+  name: string;
+  schema: SchemaOrigin;
+  tadaOutputLocation: string;
+};
+
+export async function check(
+  config: GraphQLSPConfig,
+  minSeverity: Severity = 'error'
```

Comment:
(doesn't block merging, can be a todo)

I'd argue this shouldn't be here and should be handled by the CLI. This is otherwise a little ambiguous.

At the CLI level, we could have a `--log-level/--level` option that basically just hides verbose output when it's below the specified level

Reply:
Moved it